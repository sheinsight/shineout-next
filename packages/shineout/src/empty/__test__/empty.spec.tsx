import { cleanup, render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Empty from '..';
import mountTest from '../../tests/mountTest';
import { classLengthTest } from '../../tests/structureTest';
import {
  baseTest,
  createClassName,
  displayTest,
  snapshotTest,
  textContentTest,
} from '../../tests/utils';
import EmptyBase from '../__example__/001-base';
import EmptyIcon from '../__example__/002-icon';
import EmptyImage from '../__example__/003-imge';

const SO_PREFIX = 'empty';
const originClasses = ['empty', 'wrapper', 'image', 'description'];
const originItemClasses = [] as string[];
const {
  empty,
  wrapper,
  image: imageClassName,
  description,
} = createClassName(SO_PREFIX, originClasses, originItemClasses);
afterEach(cleanup);
mountTest(<Empty />);

describe('Empty[Base]', () => {
  displayTest(Empty, 'ShineoutEmpty');
  baseTest(Empty, empty);
  snapshotTest(<EmptyBase />);
  snapshotTest(<EmptyIcon />, 'about icon');
  snapshotTest(<EmptyImage />, 'anout image');
  test('should render default', () => {
    const { container } = render(<Empty />);
    const emptyMain = container.querySelector(empty)!;
    const emptyWrapper = emptyMain.querySelector(wrapper)!;
    const emptyImage = emptyWrapper.querySelector(imageClassName)!;
    expect(emptyImage).toBeInTheDocument();
    classLengthTest(emptyImage, 'svg', 1);
    const emptyDescription = emptyWrapper.querySelector(description)!;
    expect(emptyDescription).toBeInTheDocument();
    textContentTest(emptyDescription, '暂无数据');
  });
  test('should render when set icon and description', () => {
    const { container } = render(<EmptyIcon />);
    const emptyMain = container.querySelector(empty)!;
    const emptyWrapper = emptyMain.querySelector(wrapper)!;
    const emptyImage = emptyWrapper.querySelector(imageClassName)!;
    expect(emptyImage).toBeInTheDocument();
    classLengthTest(emptyImage, 'svg', 1);
    const emptyDescription = emptyWrapper.querySelector(description)!;
    expect(emptyDescription).toBeInTheDocument();
    textContentTest(emptyDescription, '无网络');
  });
  test('should render when set icon is reactNode', () => {
    const { container } = render(<Empty icon={<div className='demo'>demo</div>} />);
    const emptyMain = container.querySelector(empty)!;
    const emptyWrapper = emptyMain.querySelector(wrapper)!;
    const emptyImage = emptyWrapper.querySelector(imageClassName)!;
    expect(emptyImage).toBeInTheDocument();
    classLengthTest(emptyImage, 'svg', 0);
    const emptyImageChildren = emptyImage.querySelector('.demo')!;
    expect(emptyImageChildren).toBeInTheDocument();
    textContentTest(emptyImageChildren, 'demo');
  });
  test('should render when set description is boolean', () => {
    const { container, rerender } = render(<Empty description={true} />);
    const emptyMain = container.querySelector(empty)!;
    const emptyWrapper = emptyMain.querySelector(wrapper)!;
    const emptyImageChildren = emptyWrapper.querySelector(description)!;
    expect(emptyImageChildren).toBeInTheDocument();
    textContentTest(emptyImageChildren, '');
    rerender(<Empty description={false} />);
    expect(emptyImageChildren).not.toBeInTheDocument();
  });
  test('should render when set imgSrc', () => {
    const { container } = render(<EmptyImage />);
    const emptyMain = container.querySelector(empty)!;
    const emptyWrapper = emptyMain.querySelector(wrapper)!;
    const emptyImage = emptyWrapper.querySelector(imageClassName)!;
    expect(emptyImage).toBeInTheDocument();
    classLengthTest(emptyImage, 'img', 1);
    const emptyDescription = emptyWrapper.querySelector(description)!;
    classLengthTest(emptyDescription, 'button', 1);
    textContentTest(emptyDescription.querySelector('button')!, 'Refresh');
  });
});
