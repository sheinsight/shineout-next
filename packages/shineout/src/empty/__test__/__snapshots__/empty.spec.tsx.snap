// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty[Base] should render correctly  1`] = `
<div
  class="so-empty-empty"
>
  <div
    class="so-empty-wrapper"
  >
    <div
      class="so-empty-image"
    >
      <svg
        fill="none"
        height="75"
        viewBox="0 0 100 75"
        width="100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          clip-path="url(#clip0_4044_10434)"
        >
          <path
            d="M100 75C100 63.3607 77.6144 53.9257 50 53.9257C22.3857 53.9257 0 63.3607 0 75H100Z"
            fill="url(#paint0_linear_4044_10434)"
          />
          <g
            filter="url(#filter0_f_4044_10434)"
            opacity="0.3"
          >
            <path
              d="M50.3087 70.0413C69.5706 70.0413 85.185 68.2375 85.185 66.0125C85.185 63.7875 69.5706 61.9835 50.3087 61.9835C31.0469 61.9835 15.4321 63.7875 15.4321 66.0125C15.4321 68.2375 31.0469 70.0413 50.3087 70.0413Z"
              fill="#979EB0"
            />
          </g>
          <path
            d="M22.6082 23.6395C22.782 23.2127 23.1957 22.9338 23.6548 22.9338H75.7168C76.1718 22.9338 76.5818 23.2071 76.7593 23.6275L82.9493 38.3482H16.6184L22.6082 23.6395Z"
            fill="#CCCFD7"
          />
          <path
            d="M75.9819 23.9172V37.2382H81.2882L75.9819 23.9172Z"
            fill="#999DA8"
          />
          <path
            d="M18.877 37.0161H81.0701C82.6314 37.0161 83.897 38.2871 83.897 39.8547V61.7805C83.897 64.2888 81.872 66.3225 79.3739 66.3225H20.5732C18.0752 66.3225 16.05 64.2888 16.05 61.7805V39.8547C16.05 38.2871 17.3157 37.0161 18.877 37.0161Z"
            fill="#E8EBF0"
          />
          <g
            filter="url(#filter1_d_4044_10434)"
          >
            <path
              d="M16.0493 42.3192C16.0493 39.8108 16.0019 37.715 20.573 37.7774H79.3737C83.5612 37.7774 83.8968 39.8108 83.8968 42.3192V61.3998C83.8968 63.9087 81.8718 65.9419 79.3737 65.9419H20.573C18.075 65.9419 16.0493 63.9087 16.0493 61.3998V42.3192Z"
              fill="black"
            />
          </g>
          <path
            d="M16.0493 42.3192C16.0493 39.8108 16.0019 37.715 20.573 37.7774H79.3737C83.5612 37.7774 83.8968 39.8108 83.8968 42.3192V61.3998C83.8968 63.9087 81.8718 65.9419 79.3737 65.9419H20.573C18.075 65.9419 16.0493 63.9087 16.0493 61.3998V42.3192Z"
            fill="#F4F5F8"
          />
          <path
            d="M23.4412 23.8855V37.0162H18.3242L23.4412 23.8855Z"
            fill="#999DA8"
          />
          <path
            d="M60.3971 50.7178H39.5502C38.0849 50.7178 36.897 51.9106 36.897 53.382C36.897 54.8534 38.0849 56.0462 39.5502 56.0462H60.3971C61.8625 56.0462 63.0501 54.8534 63.0501 53.382C63.0501 51.9106 61.8625 50.7178 60.3971 50.7178Z"
            fill="#666C7C"
          />
          <path
            d="M60.3971 49.9565H39.5502C38.0849 49.9565 36.897 51.1494 36.897 52.6208C36.897 54.0922 38.0849 55.285 39.5502 55.285H60.3971C61.8625 55.285 63.0501 54.0922 63.0501 52.6208C63.0501 51.1494 61.8625 49.9565 60.3971 49.9565Z"
            fill="#999DA8"
          />
          <path
            clip-rule="evenodd"
            d="M46.041 3.14062C42.6319 3.14062 39.8682 5.91575 39.8682 9.33899V11.4859C39.8682 14.9092 42.6319 17.6842 46.041 17.6842H54.8002C58.2093 17.6842 60.9731 14.9092 60.9731 11.4859V9.33899C60.9731 5.91575 58.2093 3.14062 54.8002 3.14062H46.041ZM47.3169 21.0086L49.386 17.6842H45.2478L47.3169 21.0086Z"
            fill="#E8EBF0"
            fill-rule="evenodd"
          />
          <path
            d="M45.6834 8.54285L47.4941 11.6922H43.8726L45.6834 8.54285Z"
            fill="#999DA8"
          />
          <path
            d="M52.304 8.54285H49.1677V11.6921H52.304V8.54285Z"
            fill="#999DA8"
          />
          <path
            d="M55.6802 11.6919C56.5463 11.6919 57.2484 10.9869 57.2484 10.1172C57.2484 9.2476 56.5463 8.5426 55.6802 8.5426C54.8142 8.5426 54.1121 9.2476 54.1121 10.1172C54.1121 10.9869 54.8142 11.6919 55.6802 11.6919Z"
            fill="#999DA8"
          />
          <path
            d="M67.7256 5.8507C66.5081 6.69694 63.5181 6.751 62.3457 7.09731L63.5875 5.64293C64.415 4.6041 64.6631 4.10547 65.6562 2.11091C66.6493 0.116362 69.3118 -0.105256 70.415 0.0332553C69.7943 1.27985 70.415 3.98081 67.7256 5.8507Z"
            fill="#E8EBF0"
          />
          <path
            d="M46.257 34.8904C45.3444 35.1307 43.5421 34.5087 42.7649 34.4595L43.824 33.8603C44.5451 33.4193 44.802 33.175 45.8298 32.198C46.8574 31.221 48.4998 31.6709 49.1305 31.9953C48.4883 32.606 48.2737 34.3593 46.257 34.8904Z"
            fill="#E8EBF0"
          />
          <path
            d="M37.262 32.7723C37.13 32.5427 37.2462 32.2485 37.4982 32.1696C39.7646 31.4593 49.3015 28.2658 55.6537 22.9713C62.0347 17.653 67.6725 11.576 69.1256 9.9791C69.3069 9.77991 69.6263 9.81841 69.76 10.0521C69.845 10.1989 69.8294 10.3803 69.7169 10.507C68.6225 11.7406 62.8044 18.1953 56.0675 23.6911C48.7371 29.6708 39.4667 32.4671 37.7105 32.9602C37.5349 33.0096 37.3532 32.9308 37.262 32.7723Z"
            fill="#E8EBF0"
          />
          <path
            d="M26.0283 27.1925C25.977 26.9505 26.1514 26.7154 26.396 26.6852C27.3291 26.5701 29.6586 26.1971 33.309 25.0577C37.011 23.9021 40.6908 21.2137 41.9723 20.2199C42.1914 20.05 42.5167 20.1675 42.5743 20.4394C42.6023 20.572 42.5585 20.7071 42.454 20.793C41.598 21.497 37.5626 24.6932 33.4811 25.8705C29.5763 26.9969 27.2876 27.3842 26.4592 27.5011C26.2561 27.5298 26.0709 27.394 26.0283 27.1925Z"
            fill="#E8EBF0"
          />
        </g>
        <defs>
          <filter
            color-interpolation-filters="sRGB"
            filterUnits="userSpaceOnUse"
            height="20.0577"
            id="filter0_f_4044_10434"
            width="81.7529"
            x="9.43213"
            y="55.9835"
          >
            <feflood
              flood-opacity="0"
              result="BackgroundImageFix"
            />
            <feblend
              in="SourceGraphic"
              in2="BackgroundImageFix"
              mode="normal"
              result="shape"
            />
            <fegaussianblur
              result="effect1_foregroundBlur_4044_10434"
              stdDeviation="3"
            />
          </filter>
          <filter
            color-interpolation-filters="sRGB"
            filterUnits="userSpaceOnUse"
            height="30.1659"
            id="filter1_d_4044_10434"
            width="67.8474"
            x="16.0493"
            y="37.776"
          >
            <feflood
              flood-opacity="0"
              result="BackgroundImageFix"
            />
            <fecolormatrix
              in="SourceAlpha"
              result="hardAlpha"
              type="matrix"
              values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
            />
            <feoffset
              dy="2"
            />
            <fecolormatrix
              type="matrix"
              values="0 0 0 0 0.803922 0 0 0 0 0.811765 0 0 0 0 0.843137 0 0 0 1 0"
            />
            <feblend
              in2="BackgroundImageFix"
              mode="normal"
              result="effect1_dropShadow_4044_10434"
            />
            <feblend
              in="SourceGraphic"
              in2="effect1_dropShadow_4044_10434"
              mode="normal"
              result="shape"
            />
          </filter>
          <lineargradient
            gradientUnits="userSpaceOnUse"
            id="paint0_linear_4044_10434"
            x1="50"
            x2="50"
            y1="53.9257"
            y2="75"
          >
            <stop
              stop-color="#E9EBEF"
            />
            <stop
              offset="1"
              stop-color="#E9EBEF"
              stop-opacity="0"
            />
          </lineargradient>
          <clippath
            id="clip0_4044_10434"
          >
            <rect
              fill="white"
              height="75"
              width="100"
            />
          </clippath>
        </defs>
      </svg>
    </div>
    <div
      class="so-empty-description"
    >
      暂无数据
    </div>
  </div>
</div>
`;

exports[`Empty[Base] should render correctly about icon 1`] = `
<div
  class="so-empty-empty"
>
  <div
    class="so-empty-wrapper"
  >
    <div
      class="so-empty-image"
    >
      <svg
        fill="none"
        height="75"
        viewBox="0 0 100 75"
        width="100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          clip-path="url(#clip0_4044_10441)"
        >
          <path
            d="M100 75C100 63.8612 77.6144 54.8319 50 54.8319C22.3857 54.8319 0 63.8612 0 75H100Z"
            fill="url(#paint0_linear_4044_10441)"
          />
          <g
            filter="url(#filter0_f_4044_10441)"
            opacity="0.3"
          >
            <path
              d="M51.1111 65.6931C57.2476 65.6931 62.2223 63.8594 62.2223 61.5966C62.2223 59.3341 57.2476 57.5 51.1111 57.5C44.9746 57.5 40 59.3341 40 61.5966C40 63.8594 44.9746 65.6931 51.1111 65.6931Z"
              fill="#979EB0"
            />
          </g>
          <path
            d="M24.8417 56.3424C24.3983 55.2501 24.9731 54.0054 26.0806 53.6597C27.188 53.314 28.3463 54.0177 28.5769 55.1764L29.4266 59.4446L26.475 60.366L24.8417 56.3424Z"
            fill="#CCCFD7"
          />
          <path
            d="M71.7444 53.7825C72.2175 52.2481 73.8825 51.4641 75.3331 52.0927C76.7837 52.7213 77.3881 54.4889 76.6356 55.9022L73.865 61.1093L70 59.4341L71.7444 53.7825Z"
            fill="#D4D9E5"
          />
          <path
            d="M22.0035 59.1355C21.4287 58.5386 21.4827 57.5656 22.1199 57.0386C22.757 56.5121 23.7001 56.6604 24.1525 57.3588L25.819 59.9302L24.1208 61.3344L22.0035 59.1355Z"
            fill="#D4D9E6"
          />
          <path
            d="M52.4691 42.5H50V52.5841H52.4691V42.5Z"
            fill="#CCCFD7"
          />
          <path
            d="M44.0059 35.0044L45.6047 33.3654L52.0001 39.9216L50.4012 41.5607L44.0059 35.0044Z"
            fill="url(#paint1_linear_4044_10441)"
          />
          <path
            d="M43.75 34.7641L45.3488 33.125L51.7443 39.6813L50.1454 41.3203L43.75 34.7641Z"
            fill="#CCCFD7"
          />
          <path
            d="M51.0111 43.386C52.2599 43.386 53.2723 42.3483 53.2723 41.068C53.2723 39.7878 52.2599 38.75 51.0111 38.75C49.7624 38.75 48.75 39.7878 48.75 41.068C48.75 42.3483 49.7624 43.386 51.0111 43.386Z"
            fill="#B3B7C1"
          />
          <path
            d="M49.832 12.3694C55.223 19.9986 54.5824 31.3696 48.167 37.535C41.7516 43.7004 31.1315 43.7004 23.6917 37.4913L49.832 12.3694Z"
            fill="url(#paint2_linear_4044_10441)"
          />
          <path
            d="M49.832 12.3694C55.223 19.9986 54.5824 31.3696 48.167 37.535C41.7516 43.7004 31.1315 43.7004 23.6917 37.4913L49.832 12.3694Z"
            fill="#E8EBF0"
          />
          <path
            d="M41.2931 29.8931C48.5507 22.9831 52.4648 15.2252 50.0354 12.5653C47.606 9.90539 39.7532 13.3508 32.4956 20.2608C25.2381 27.1707 21.324 34.9286 23.7534 37.5886C26.1828 40.2485 34.0355 36.8031 41.2931 29.8931Z"
            fill="#F4F5F8"
          />
          <path
            d="M40.9881 29.2713C50.1128 20.891 52.0853 14.4286 49.888 12.0003C47.6907 9.57183 40.0516 13.2329 32.8255 20.1774C25.5994 27.1219 21.5229 34.7201 23.7201 37.1485C25.9175 39.577 33.4892 36.3884 40.9881 29.2713Z"
            fill="#999DA8"
          />
          <path
            clip-rule="evenodd"
            d="M31.417 17.6481L42.3901 27.8798L40.19 30.147L30.2615 18.8378C29.9686 18.5041 29.9811 17.9951 30.29 17.6768C30.5972 17.3601 31.0946 17.3474 31.417 17.6481Z"
            fill="#F4F5F8"
            fill-rule="evenodd"
          />
          <path
            d="M29.2 18.8614C30.4887 18.8614 31.5334 17.7848 31.5334 16.4566C31.5334 15.1284 30.4887 14.0518 29.2 14.0518C27.9112 14.0518 26.8665 15.1284 26.8665 16.4566C26.8665 17.7848 27.9112 18.8614 29.2 18.8614Z"
            fill="#B3B7C1"
          />
          <path
            d="M29.9876 7.50656C30.6284 7.43781 31.203 7.91743 31.271 8.57787C31.339 9.23837 30.8746 9.8295 30.2338 9.89825L30.2349 9.90912C30.1061 9.91562 29.9768 9.92587 29.847 9.93975C26.0019 10.3524 23.2156 13.8992 23.6235 17.8619C23.6799 18.41 23.7945 18.9372 23.9602 19.438L23.9198 19.4421C23.9522 20.0761 23.4975 20.6312 22.8776 20.6978C22.2368 20.7666 21.6622 20.2869 21.5941 19.6264L21.6027 19.6904L21.5952 19.6914C21.4575 19.182 21.3588 18.6541 21.3029 18.1109C20.759 12.8274 24.474 8.09825 29.6008 7.54806C29.6965 7.53781 29.792 7.52906 29.8874 7.52181L29.905 7.5185L29.9876 7.50656Z"
            fill="#CCCFD7"
          />
          <path
            d="M24.1625 3.28992L24.1657 3.33396C24.7808 3.27947 25.3256 3.72523 25.4215 4.34306L25.4317 4.42834C25.4884 5.08989 25.0141 5.67267 24.3721 5.73002L24.3747 5.76503C19.7959 7.5713 16.7399 12.3378 17.1929 17.6247C17.2512 18.3055 17.3654 18.9675 17.5304 19.6064L17.5084 19.6082C17.5651 20.2698 17.0907 20.8525 16.4487 20.9099C15.8069 20.9672 15.2405 20.4774 15.1838 19.8159L15.1766 19.8163C15.0303 19.1724 14.9264 18.51 14.8683 17.8324C14.3011 11.2134 18.2911 5.27034 24.1625 3.28992Z"
            fill="#E8EBF0"
          />
          <path
            d="M34.1233 1.24378L32.1232 3.30428L34.1234 5.36564C34.4003 5.6511 34.4003 6.11111 34.1234 6.39656C33.8498 6.6785 33.4041 6.68069 33.1279 6.40137C33.1263 6.39969 33.1248 6.39806 33.1232 6.3965L31.1233 4.33472L29.124 6.396C28.8504 6.678 28.4048 6.68025 28.1285 6.40088C28.1269 6.39931 28.1253 6.39769 28.1238 6.39612C27.8468 6.11063 27.8468 5.65061 28.1238 5.36516L30.123 3.30476L28.1233 1.24387C27.8463 0.958419 27.8463 0.498404 28.1233 0.212946C28.3969 -0.0690643 28.8427 -0.0711893 29.1189 0.2082C29.1204 0.20979 29.122 0.211388 29.1236 0.212994L31.1228 2.27383L33.123 0.212898C33.3967 -0.0691393 33.8425 -0.0712218 34.1188 0.208246C34.1203 0.209806 34.1218 0.211372 34.1234 0.212946C34.4003 0.498388 34.4003 0.958388 34.1233 1.24378Z"
            fill="#CCCFD7"
          />
          <path
            d="M50.9587 47.8992C55.5792 47.8992 59.3248 51.739 59.3248 56.4757C59.3248 57.0304 59.2735 57.5728 59.1754 58.0981H42.7421C42.6439 57.5728 42.5925 57.0304 42.5925 56.4757C42.5925 51.739 46.3382 47.8992 50.9587 47.8992Z"
            fill="#CCCFD7"
          />
          <path
            d="M59.1064 58.125H42.8392C41.9616 58.125 41.25 58.8514 41.25 59.7476C41.25 60.6437 41.9616 61.3702 42.8392 61.3702H59.1064C59.9841 61.3702 60.6956 60.6437 60.6956 59.7476C60.6956 58.8514 59.9841 58.125 59.1064 58.125Z"
            fill="#B3B7C1"
          />
        </g>
        <defs>
          <filter
            color-interpolation-filters="sRGB"
            filterUnits="userSpaceOnUse"
            height="20.1931"
            id="filter0_f_4044_10441"
            width="34.2222"
            x="34"
            y="51.5"
          >
            <feflood
              flood-opacity="0"
              result="BackgroundImageFix"
            />
            <feblend
              in="SourceGraphic"
              in2="BackgroundImageFix"
              mode="normal"
              result="shape"
            />
            <fegaussianblur
              result="effect1_foregroundBlur_4044_10441"
              stdDeviation="3"
            />
          </filter>
          <lineargradient
            gradientUnits="userSpaceOnUse"
            id="paint0_linear_4044_10441"
            x1="50"
            x2="50"
            y1="54.8319"
            y2="75"
          >
            <stop
              stop-color="#E9EBEF"
            />
            <stop
              offset="1"
              stop-color="#E9EBEF"
              stop-opacity="0"
            />
          </lineargradient>
          <lineargradient
            gradientUnits="userSpaceOnUse"
            id="paint1_linear_4044_10441"
            x1="525.323"
            x2="504.891"
            y1="33.3654"
            y2="812.688"
          >
            <stop
              stop-color="#D1EAFF"
            />
            <stop
              offset="1"
              stop-color="#85C9FF"
            />
          </lineargradient>
          <lineargradient
            gradientUnits="userSpaceOnUse"
            id="paint2_linear_4044_10441"
            x1="2386.56"
            x2="1802.2"
            y1="1108.32"
            y2="2930.71"
          >
            <stop
              stop-color="#D1EAFF"
            />
            <stop
              offset="1"
              stop-color="#85C9FF"
            />
          </lineargradient>
          <clippath
            id="clip0_4044_10441"
          >
            <rect
              fill="white"
              height="75"
              width="100"
            />
          </clippath>
        </defs>
      </svg>
    </div>
    <div
      class="so-empty-description"
    >
      无网络
    </div>
  </div>
</div>
`;

exports[`Empty[Base] should render correctly anout image 1`] = `
<div
  class="so-empty-empty"
>
  <div
    class="so-empty-wrapper"
  >
    <div
      class="so-empty-image"
    >
      <img
        alt="empty"
        src="https://raw.githubusercontent.com/sheinsight/shineout-static/main/shineout-next/images/image/s-02.png"
      />
    </div>
    <div
      class="so-empty-description"
    >
      <button
        class="so-button-primary so-button-button"
      >
        <span>
          Refresh
        </span>
      </button>
    </div>
  </div>
</div>
`;
